src/storage.h

```diff
@@ -0,0 +1,253 @@
+class Storage 
+  : public cyclus::Facility,
+    public cyclus::toolkit::CommodityProducer {
+ public:  
+  /// @param ctx the cyclus context for access to simulation-wide parameters
+  Storage(cyclus::Context* ctx);
+  
+  #pragma cyclus decl
+
+  #pragma cyclus note {"doc": "Storage is a simple facility which accepts any number of commodities " \
+                              "and holds them for a user specified amount of time. The commodities accepted "\
+                              "are chosen based on the specified preferences list. Once the desired amount of material "\
+                              "has entered the facility it is passed into a 'processing' buffer where it is held until "\
+                              "the residence time has passed. The material is then passed into a 'ready' buffer where it is "\
+                              "queued for removal. Currently, all input commodities are lumped into a single output commodity. "\
+                              "Storage also has the functionality to handle materials in discrete or continuous batches. Discrete "\
+                              "mode, which is the default, does not split or combine material batches. Continuous mode, however, "\
+                              "divides material batches if necessary in order to push materials through the facility as quickly "\
+                              "as possible."}
+
+  /// A verbose printer for the Storage Facility
+  virtual std::string str();
+
+  // --- Facility Members ---
+  
+  // --- Agent Members ---
+  /// Sets up the Storage Facility's trade requests
+  virtual void EnterNotify();
+
+  /// The handleTick function specific to the Storage.
+  virtual void Tick();
+
+  /// The handleTick function specific to the Storage.
+  virtual void Tock();
+
+  /* --- Storage Members --- */
+
+  /// @brief the minimum processing time required for a full process
+  inline void residence_time_(int t) { residence_time = t; }
+  inline int residence_time_() const { return residence_time; }
+
+  /// @brief the maximum amount allowed in inventory
+  inline void max_inv_size_(double c) { max_inv_size = c; }
+  inline double max_inv_size_() const { return max_inv_size; }
+
+  /// @brief the maximum amount processed per timestep
+  inline void throughput_(double c) { throughput = c; }
+  inline double throughput_() const { return throughput; }
+
+  /// @brief the in commodity
+  inline void in_commods_(std::string c) { in_commods.push_back(c); }
+  inline std::vector<std::string> in_commods_() const { return in_commods; }
+
+  /// @brief the in commodity preferences
+  inline void in_commod_prefs_(double c) { in_commod_prefs.push_back(c); }
+  inline double in_commod_prefs_() const { return in_commod_prefs.front(); }
+
+  /// @brief the out commodity
+  inline void out_commods_(std::string c) { out_commods.push_back(c); }
+  inline std::string out_commods_() const {
+    if (out_commods.size() == 0){
```

Comment:
I don't think this is conventional behavior.  If the user doesn't set the out_commod, then we should not return anything here.  This is especially true since this is used to initalize things,